compiler/rustc_hir_analysis/src/variance/constraints.rs

```diff
@@ -127,11 +131,16 @@ impl<'a, 'tcx> ConstraintContext<'a, 'tcx> {
             }
 
             ty::Error(_) => {}
+
             _ => {
-                span_bug!(
-                    tcx.def_span(def_id),
-                    "`build_constraints_for_item` unsupported for this item"
-                );
+                if tcx.features().lazy_type_alias && let DefKind::TyAlias = tcx.def_kind(def_id) {
+                    self.add_constraints_from_ty(current_item, ty, self.covariant);
```

Comment:
Leave a comment explaining that we need to check the `DefKind`, because `type_of(def_id)` returns the type behind the alias, not `ty::Alias(ty::Weak, ..)`